nullaway/src/main/java/com/uber/nullaway/handlers/AbstractFieldContractHandler.java

```diff
@@ -0,0 +1,280 @@
+   *
+   * @return Returns true, if the annotation conforms to the syntax rules.
+   */
+  protected boolean validateAnnotationSyntax(
+      Set<String> content,
+      NullAway analysis,
+      MethodTree tree,
+      VisitorState state,
+      Symbol.MethodSymbol methodSymbol) {
+    if (content == null) {
+      return true;
+    } else {
+      if (content.isEmpty()) {
+        // we should not allow useless annotations.
+        reportMatch(
+            analysis,
+            state,
+            tree,
+            new ErrorMessage(
+                ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID,
+                "empty @"
+                    + annotName
+                    + " is the default precondition for every method, please remove it."));
+        return false;
+      } else {
+        for (String fieldName : content) {
+          if (fieldName.contains(".")) {
+            if (!fieldName.startsWith(THIS_NOTATION)) {
+              reportMatch(
+                  analysis,
+                  state,
+                  tree,
+                  new ErrorMessage(
+                      ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID,
+                      "currently @"
+                          + annotName
+                          + " supports only class fields of the method receiver: "
+                          + fieldName
+                          + " is not supported"));
+              return false;
+            } else {
+              fieldName = trimReceiver(fieldName);
+            }
+          }
+          Symbol.ClassSymbol classSymbol = ASTHelpers.enclosingClass(methodSymbol);
+          Element field = getFieldFromClass(classSymbol, fieldName);
+          if (field == null) {
+            reportMatch(
+                analysis,
+                state,
+                tree,
+                new ErrorMessage(
+                    ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID,
+                    "cannot find field ["
+                        + fieldName
+                        + "] in class: "
+                        + classSymbol.getSimpleName()));
+            return false;
+          }
+          if (field.getModifiers().contains(Modifier.STATIC)) {
+            reportMatch(
+                analysis,
+                state,
+                tree,
+                new ErrorMessage(
+                    ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID,
+                    "cannot accept static field: ["
+                        + fieldName
+                        + "] as a parameter in @"
+                        + annotName
+                        + " annotation"));
+            return false;
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns the field name excluding its receiver (e.g. "this.a" will be "a")
+   *
+   * @param fieldName A class symbol.
+   * @return The class field name.
+   */
+  public static String trimReceiver(String fieldName) {
```

Comment:
Can we move these `static` methods to a utility class?  I find the method references to this method written `EnsuresNonNullHandler::trimReceiver` and `RequiresNonNullHandler::trimReceiver` to be confusing.  Also, if we put the utility class in the same package, the methods can have package (default) visibility rather than being `public`